nixos/modules/services/audio/mpdscheduler.nix

```diff
@@ -0,0 +1,59 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let 
+  cfg = config.services.mpdscheduler;
+in {
+  options.services.mpdscheduler = {
+    enable = mkEnableOption "mpdscheduler service";
+    host = mkOption {
+      type = types.str;
+      description = "the mpd host to connect to";
+      default = "localhost";
+      example = "mpdHost.local";
+    };
+    port = mkOption {
+      type = types.int;
+      description = "the mpd tcp port to connect to";
+      default = 6600;
+    };
+    additionalChannels = mkOption {
+      type = types.bool;
+      default = true;
+      description = "whether the 'alarm', 'sleep' and 'cancel' channels shall be used in addition to the 'scheduler' channel";
+    };
+    fadeTime = mkOption {
+      type = types.int;
+      default = 30;
+      description = "alarm/sleep fade time in seconds";
+      example = 15;
+    };
+    maxVolume = mkOption {
+      type = types.int;
+      default = 100;
+      description = "the volume to wake at in percent";
+      example = 50;
+    };
+    waitForMpd = mkOption {
+      type = types.bool;
+      default = true;
+      description = "whether to wait for mpd to start before starting, set to false if mpdscheduler is connecting to mpd on another machine";
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.mpdscheduler = {
+      wantedBy = [ "multi-user.target" ];
+      environment = {
+        MPD_HOST = "${cfg.host}";
+        MPD_PORT = "${toString cfg.port}";
+        MPDSCHEDULER_FADE_TIME = "${toString cfg.fadeTime}";
+        MPDSCHEDULER_MAX_VOLUME = "${toString cfg.maxVolume}";
+      };
+      after = mkIf cfg.waitForMpd [ "mpd.service" ];
```

Comment:
Personally I like `after = mkIf (cfg.host == "localhost") [ "mpd.service" ];` more.

Reply:
I was considering this, but there are other players like `mopidy` that provide the same interface, therefore having `cfg.host == "localhost"` does not imply using `mpd` - on the other hand there is no such option for `mopidy`, so I guess I should include an option for that as well. or maybe have an option like `waitForPlayer` that points to a compatible player...

Reply:
I want aware of this. I still might suggest the same thing I said and let the user override if the use another client (the software is called `mpdscheduler` afterall) or... not sure, looking forward to see what you decide.

Reply:
I went with your suggestion for now.